activerecord/test/models/cpk/order.rb

```diff
@@ -6,6 +6,6 @@ class Order < ActiveRecord::Base
     self.primary_key = [:shop_id, :id]
 
     has_many :order_agreements, primary_key: :id
-    has_many :books
+    has_many :books, primary_key: :id
```

Comment:
Why do we need to add `primary_key: :id` here? All the tests appear to pass without it. Same applies for book.rb

Reply:
Not strictly required, I initially setup a test that had a uniqueness constraint on the association, after trying to associate a book to an order like:

```ruby
p book_one # => Book(author_id: 1, number: 2, revision: 1, order_id: nil)
p order # => Order(shop_id: 1, id: 3)
order.books << book_one
p book_one # => Book(author_id: 1, number: 2, revision: 1, order_id: 1)
```

In other words, it was incorrectly using the `shop_id`, pointing Active Record to using the `id` fixed the problem.

You're right, though, this isn't used right now.